### PR TITLE
DAOS-16584 test: Support running independent io sys admin steps

### PR DESCRIPTION
**Pull request by daltonbohning**
Support running independent io sys admin steps from the yaml.

Skip-unit-tests: true
Skip-fault-injection-test: true

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/deployment/io_sys_admin.yaml [hunk 1]
```diff
@@ -104,3 +104,11 @@ dcp:
...
 hdf5_vol:
   plugin_path: /usr/lib64/mpich/lib
+
+io_sys_admin:
+  # Toggle independent steps
+  run_step_pool_create_ownership: True
+  run_step_storage_system_query: True
+  run_step_io: True
+  run_step_snapshot: True
+  run_test_datamover: True
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/io_sys_admin.yaml:114]:
```suggestion
  run_step_datamover: True
```

File: src/tests/ftest/deployment/io_sys_admin.py [hunk 2]
```diff
@@ -40,66 +40,87 @@ def test_io_sys_admin(self):
...
+                        nvme_free_space_before_snap_destroy)
+                    self.log.info(
+                        "Free space when test terminated: %s", self.get_free_space()[1])
+                    self.fail("Aggregation did not complete as expected")
+
+                time.sleep(60)
+                returned_space = self.get_free_space()[1] - nvme_free_space_before_snap_destroy
+                counter += 1
+
+        if run_test_datamover:
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/io_sys_admin.py:115]:
```suggestion
        if run_step_datamover:
```

**Comment by daltonbohning** [comment]
From the steps.log, all 5 steps are being executed
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15134/4/artifact/Functional%20Hardware%20Medium/deployment/io_sys_admin.py/steps.log
```
2024-09-17 18:21:56,813 test             L0471 INFO | ==> Step 4: Verify pool creation time and container set-owner [elapsed since last step: 0.01s]
2024-09-17 18:23:11,398 test             L0471 INFO | ==> Step 5: Verify storage scan and system query [elapsed since last step: 74.59s]
2024-09-17 18:23:17,405 test             L0471 INFO | ==> Step 6: Verifying large dataset IO [elapsed since last step: 6.01s]
2024-09-17 18:25:30,310 test             L0471 INFO | ==> Step 7: Verifying snapshot creation and aggregation [elapsed since last step: 132.90s]
2024-09-17 18:26:37,771 test             L0471 INFO | ==> Step 8: Verifying datamover [elapsed since last step: 67.46s]

```

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]
Maybe we can separate them to five different tests? Anyway, this is much better. Thanks.

**Comment by daltonbohning** [comment]
> Maybe we can separate them to five different tests? Anyway, this is much better. Thanks.

Unfortunately some of the steps rely on previous steps for the pool and container to be created. We could just have each step start fresh in a separate test case, but then we can't run all 5 tests in sequential order because avocado doesn't guarantee sorted runs

**Comment by phender** [comment]
> > Maybe we can separate them to five different tests? Anyway, this is much better. Thanks.
> 
> Unfortunately some of the steps rely on previous steps for the pool and container to be created. We could just have each step start fresh in a separate test case, but then we can't run all 5 tests in sequential order because avocado doesn't guarantee sorted runs

For different test files that's true, but if we defined separate test methods in the same file they would run in order.  The problem really is that while we can keep the server and agent up and running across test variants, we currently always destroy the pools and containers.

**Comment by daltonbohning** [comment]
> For different test files that's true, but if we defined separate test methods in the same file they would run in order. The problem really is that while we can keep the server and agent up and running across test variants, we currently always destroy the pools and containers.

Methods in the SAME file DO run in order?

**Pull request merged**